### PR TITLE
CYBERNETICS : cyberlinks [Atomized part 1 / 3]

### PR DESCRIPTION
**Pull request by Djiq**
## About The Pull Request

just a note, i dont look at prs as often as i used to due to my other more important projects.

Current plan : 

1. Cyberlinks (Pain in the ass as this is the balancing mechanism for point 3) WE ARE HERE
2. Hacking (Useless without point 3)
3. Blackmarket + implants (Too op to PR first, so it goes last, for this stage to be possible 2 previous prs must be merged)

CYBERLINKS

    NT Cyberlink 1.0
    NT Cyberlink 2.0
    Terran Cyberware System
    Cybersun Cybernetics Access System

Aight so a little explanation here, what's up with all of those cyberlinks you may ask, well they are a way of gatekeepin certain implants, so you cannot effortlessly just have a lot of OP shit on you. the way they work is that, every implant has something called 'PROTOCOLS', each cyberlink also has PROTOCOLS, if your implant doesn't share even 1 PROTOCOL for every PROTOCOL_CLASS it will be uncompatible and will not function.

why im actually reviving this now and not later? i got bored this afternoon, and hey im missing a side project i may as well continue this.

More explanation : https://github.com/tgstation/tgstation/pull/55936

## Why It's Good For The Game

look original pr : https://github.com/tgstation/tgstation/pull/55936

After playing Cyberpunk i realized our game lacked a lot of cybernetics, and fun that comes from them. The way these are coded pushes people to work together, since implants must be installed by another person, unless someone is fully augmented.

Not only that but a lot of these items are very rare to find, and have only a tiny chance of them appearing, along with a hefty price, meaning unless you are a very lucky head of staff, you will have to interact with the current economy system to get these implants. Which is a positive addtion to the game.

## Changelog
:cl:
add: Cyberlinks - gatekeeping implants
add: NT Cyberlink 1.0
add: NT Cyberlink 2.0
add: Terran Cyberware System
add: Cybersun Cybernetics Access System
/:cl:

### CODE REVIEW

**Comment by Jared-Fogle** [comment]
> Adds cyberlinks, explanation on what they are located in original pr : #55936

Please include the relevant details in your PR body, so we don't have to go digging for it.

**Comment by EdgeLordExe** [comment]
> 
> 
> > Adds cyberlinks, explanation on what they are located in original pr : #55936
> 
> Please include the relevant details in your PR body, so we don't have to go digging for it.

lemme copy paste it

**Comment by EdgeLordExe** [comment]
i have no clue why this fails the tests

**Comment by ATH1909** [comment]
it lives on!

**Comment by davethwave** [comment]
Thanks for the hard work, if anything I hope this will be able to pass now and get merged.

**Comment by Investigator77** [comment]
Legendary

**Comment by EdgeLordExe** [comment]
i still have no clue why this runtimes

File: code/__DEFINES/atom_hud.dm [hunk 1]
```diff
@@ -65,21 +65,21 @@
...
-#define ANTAG_HUD_NINJA 		17
-#define ANTAG_HUD_CHANGELING 	18
-#define ANTAG_HUD_ABDUCTOR 		19
-#define ANTAG_HUD_BROTHER		20
-#define ANTAG_HUD_OBSESSED		21
-#define ANTAG_HUD_FUGITIVE		22
-#define ANTAG_HUD_GANGSTER		23
-#define ANTAG_HUD_SPACECOP		24
-#define ANTAG_HUD_HERETIC		25
+#define ANTAG_HUD_CULT			12
```
Comment (TheGamerdk) [inline code/__DEFINES/atom_hud.dm:68]: me thinks the skipped hud 11 is causing your runtimes

Comment (EdgeLordExe) [reply code/__DEFINES/atom_hud.dm:68]: maybe

**Comment by EdgeLordExe** [comment]
oh this actually worked lol

**Comment by TheGamerdk** [comment]
:)

**Comment by EdgeLordExe** [comment]
Hello this compiles, review please

**Comment by Fikou** [comment]
changelog should explain what a cyberlink is imo

**Comment by EdgeLordExe** [comment]
reminder: cybernetics are job content for roboticists mainly, and anyone else that bothers to interact with economy, if you dont want cybernetics just fucking tell me, i wont be upset. Keeping this PR is just a waste of time, especially when i'm given the silent treatment.

**Comment by davethwave** [comment]
personally I'm looking forward to it myself. Hope others can be more vocal about it as well.

**Comment by interestingusernam3** [comment]
personally I'm looking forward to it myself. Hope others can be more vocal about it as well.

**Comment by EdgeLordExe** [comment]
this is pointless

**Comment by Fikou** [comment]
why close this

**Comment by GunHog** [comment]
Please do not let this die! You are likely frustrated due to not having been reviewed yet, but you have to hang in there! This PR is quite huge balance-wise, and will be a lot of work to look through and test all of its content. It is annoying to have to keep the code updated against merge conflicts and from going stale while you await a review. Please do keep at it though. Every PR asks a maintainer, who does this work for free, for a slice of his personal time and effort. This content is worth keeping alive!

**Comment by UnokiAs** [comment]
I miss this 😔

**Comment by EdgeLordExe** [comment]
> I miss this pensive

then go play on some better server, i aint gonna work for tg unless i get paid or something. Trying to get anything in here merged with the current GBP-crazy maintainers and the lack of response that was well demonstrated in this PR is a nightmare, sincerely edge.

**Comment by SmArtKar** [comment]
> > I miss this pensive
> 
> then go play on some better server, i aint gonna work for tg unless i get paid or something. Trying to get anything in here merged with the current GBP-crazy maintainers and the lack of response that was well demonstrated in this PR is a nightmare, sincerely edge.

is it fine to continue your work on cybernetics then?

**Comment by EdgeLordExe** [comment]
> > > I miss this pensive
> > 
> > 
> > then go play on some better server, i aint gonna work for tg unless i get paid or something. Trying to get anything in here merged with the current GBP-crazy maintainers and the lack of response that was well demonstrated in this PR is a nightmare, sincerely edge.
> 
> is it fine to continue your work on cybernetics then?

yes, obviously. it's AGPL3.0 you're free to do anything with that code, i promised myself not to code anything for TG due to how i was treated during the making of this PR. After i left i ported this to merchant and they merged it, here is the link to the pr on merchantcode: https://github.com/MisterBones2910/Merchan-TG-uild/pull/2

**Pull request closed without merging**